Client/Impl/Misc/PersistedAccessTokenCache.cs

```diff
@@ -70,12 +81,26 @@ private async Task<string> GetValidToken(string audience, AccessToken currentAcc
     {
         var now = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
         var dueDate = currentAccessToken.DueDate;
-        if (now < dueDate)
+
+        if (now < dueDate - 15000) // Adding 15 sec buffer to update the token before it really expires.
```

Comment:
Lets focus on one topic/part in this PR. We can do/fix this with https://github.com/camunda-community-hub/zeebe-client-csharp/issues/642